### ISSUE TITLE
Bug: [no-restricted-imports] handle core rule changes

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#11889]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

n/a

### Repro Code

```TypeScript
n/a
```

### ESLint Config

```javascript
n/a
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

n/a

### Actual Result

n/a

### Additional Info

https://github.com/eslint/eslint/pull/19562 appears to have broken some of our tests; see https://github.com/typescript-eslint/typescript-eslint/pull/11691

We should
1. Determine whether the extension rule's behavior needs to be changed and make changes if necessary
1. Determine whether the extensions rule's behavior should be core-version-dependent and deal with that if necessary
1. Fix broken tests.

**Comment by rami0617** [comment]
Hello, I'm looking into this, can I take it?

**Comment by kirkwaiblinger** [comment]
Looks like the way that this rule is composed with the base rule, it's somewhat resilient to the changes in the base rule... So looking at now, it's not obvious that there's any code changes that we need to make in order to be compatible with the base rule (and a few test changes have already been made, so that part of this issue is obsolete). 

I think the easiest way from here would be to mark this rule as deprecated in favor of ESLint's version, and with intention to remove it once typescript-eslint's minimum ESLint peer dep includes https://github.com/eslint/eslint/pull/19562 ?

**Comment by beanscg** [comment]
Bean Labs took a public-source pass on this issue.

The narrowest useful next step looks like: map upstream option compatibility

Public evidence checked:
- src/
- docs/

A good acceptance check would be a docs diff that changes only the affected wording/link and renders without stale references.

No PR opened from this pass.
